src/Munee/Asset/Type/Css.php

```diff
@@ -157,107 +159,105 @@ protected function isScss($file)
     }
 
     /**
-     * Fixes relative paths to absolute paths
+     * Use CssParser to go through and convert all relative paths to absolute
      *
-     * @param $content
-     * @param $originalFile
+     * @param string $content
+     * @param string $originalFile
      *
      * @return string
-     *
-     * @throws CompilationException
      */
-    protected function fixRelativeImagePaths($content, $originalFile)
+    protected function fixRelativePaths($content, $originalFile)
     {
-        $regEx = '%(url[\\s]*\\()(?!data:image)[\\s\'"]*([^\\)\'"]*)[\\s\'"]*(\\))%';
+        $cssParser = new CssParser($content);
+        $cssDocument = $cssParser->parse();
 
-        $webroot = $this->request->webroot;
-        $changedContent = preg_replace_callback($regEx, function ($match) use ($originalFile, $webroot) {
-            $filePath = trim($match[2]);
-            // Skip conversion if the first character is a '/' since it's already an absolute path
-            // Also skip conversion if the string has an protocol in url
-            if ($filePath[0] !== '/' && strpos($filePath, '://') === false) {
-                $basePath = SUB_FOLDER  . str_replace($webroot, '', dirname($originalFile));
-                $basePathParts = array_reverse(array_filter(explode('/', $basePath)));
-                $numOfRecursiveDirs = substr_count($filePath, '../');
-                if ($numOfRecursiveDirs > count($basePathParts)) {
-                    throw new CompilationException(
-                        'Error in stylesheet <strong>' . $originalFile .
-                        '</strong>. The following URL goes above webroot: <strong>' . $filePath .
-                        '</strong>'
-                    );
-                }
+        $cssBlocks = $cssDocument->getAllValues();
 
-                $basePathParts = array_slice($basePathParts, $numOfRecursiveDirs);
-                $basePath = implode('/', array_reverse($basePathParts));
+        $this->fixUrls($cssBlocks, $originalFile);
 
-                if (! empty($basePath) && $basePath[0] != '/') {
-                    $basePath = '/' . $basePath;
+        return $cssDocument->render();
```

Comment:
Not always no. In function calls and foreach loops you generall have to do `&` unless the object copy does something smart/funny.
```php
php >$ar = array(1,2,3,4);
php > foreach ($ar as $val) {
php { $val = 1;
php { }
php > print_r($ar);
Array
(
    [0] => 1
    [1] => 2
    [2] => 3
    [3] => 4
)```